### ISSUE TITLE
ScaleBar is always one vb state behind

### ISSUE THREAD

**Issue by sem-geologist** [pyqtgraph/pyqtgraph#437]
I am trying to add scalebar to my image as in one of pyqtgraph examples. Firstly I noticed that in my plot scale bar was not appearing directly after I added it to the vb. I checked example and behaviour there is the same - scale bar appears only if plot is moved around. But that would be false bug report. The real behavior is understood only after doing scroll zooming in and out (it is same on the working pyqtgraph example). Particularly pay attention to the first scale change after first scroll direction/zoom change. Only then it is clear that scalebar shows not actual vb scale but one step before actual!
I think this is a bug, I was looking through the source, but can't find exactly where it propagates from. 

python3, both Qt5 and Qt4 backends.

**Comment by sjmvm** [comment]
I can confirm the issue exists for python 3.4 - PySide 1.2.4

I couldn't find the bug either, but I suspect it has something to do with the viewbox rescaling only after the sigRangeChanged has been emited, as the emitted state still reflects the old range. There is a simple work-around: use a QTimer to delay the update of the scaleBar. I believe there must be more elegant approaches - capturing the signal of the ViewBox when the range has effectively changed - , but this one works for me.
```
    def parentChanged(self):
        view = self.parentItem()
        if view is None:
            return
        view.sigRangeChanged.connect(self.updateDelay)
        self.updateDelay()

    def updateDelay(self):
        QtCore.QTimer.singleShot(100, self.updateBar)

    def updateBar(self):
        view = self.parentItem()
        if view is None:
            return
        p1 = view.mapFromViewToItem(self, QtCore.QPointF(0,0))
        p2 = view.mapFromViewToItem(self, QtCore.QPointF(self.size,0))
        w = (p2-p1).x()
        self.bar.setRect(QtCore.QRectF(-w, 0, w, self._width))
        self.text.setPos(-w/2., 0)
```

**Comment by sem-geologist** [comment]
Interestingly on python3.8, and pyqtgraph 0.12.2 I see no more of this behaviour, can't anymore replicate or find what changes in pyqtgraph fixed this. Probably it is some Qt changes which fixed this. Thus I am closing this.

**Comment by j9ac9k** [comment]
Hi @sem-geologist just wanted to say thanks for closing out an issue from this long ago.  As our PR count dwindles, the next thing we're going to turn our attention to is the issue tracker, having authors close issues that are no longer relevant is a big help 👍🏻
